src/vs/platform/actions/browser/toolbar.ts

```diff
@@ -154,8 +154,15 @@ export class WorkbenchToolBar extends ToolBar {
 		// count for max
 		if (this._options?.overflowBehavior !== undefined) {
 
-			const exempted = new Set(this._options.overflowBehavior.exempted);
-			const maxItems = this._options.overflowBehavior.maxItems - exempted.size;
+			const primaryIds = new Set(primary.map(a => a.id));
+			const exemptedIds = new Set(this._options.overflowBehavior.exempted);
+			let exemptedCount = 0;
+			for (const id of primaryIds) {
+				if (exemptedIds.has(id)) {
+					exemptedCount++;
+				}
+			}
```

Comment:
@jrieken this was needed because not all exempted commands are visible at the same time, so we need to figure out which ones are really showing.

Reply:
So, its basically the intersection of exempted and actual commands, right?

Reply:
@jrieken yes!

Reply:
Oh, cool. I got snipped and added an intersection util and adopted it here. Hope you don't kind

Reply:
👏